src/generic_mapper.xml

```diff
@@ -0,0 +1,1258 @@
+            must be located in profile folder
+
+
+        Mapping Aliases
+        map mode &lt;simple, normal, or complex&gt; - sets the mapping mode, determining what
+            exits are set automatically as you move around
+        set area &lt;area name&gt; - move the current room into the named area, area will be
+            created if it does not currently exist
+        shift &lt;direction&gt; - use this command to move the room you are currently in around
+            on the map
+        merge rooms - use this command to combine the room you are in with any other rooms
+            in the same location and with the same name
+        clear moves - use this command to clear the move queue after you attempt a move
+            that doesn't succeed, but for which there is no trigger indicating this with
+            the onMoveFail event
+        add door &lt;direction&gt; &lt;optional none, open, closed, locked&gt; &lt;optional yes, no&gt; -
+            adds a door in the given direction, defaulting to closed (use none to remove
+            a door), and defaulting 'no' for one-way status
+        add portal &lt;entry command&gt; - adds a portal that uses the given command for entry
+            to force this command to create a new room, rather than linking to an existing
+            one, put -f before the entry command, like this: add portal -f enter portal
+        set exit &lt;direction&gt; &lt;roomID&gt; - sets the given direction to connect, one way,
+            to the room with the specified roomID, used for very complex areas
+            to set the destination room for a portal, put -p before the portal entry command
+            like this: set exit -p enter portal 1124
+
+        Normal User Aliases
+        find me - use this command to search the entire map to try to locate you based on
+            room name and exits, typically not necessary, as this will be done anyway if
+            a person moves and their location is unknown
+        find path &lt;room name&gt; OR &lt;room name&gt; ; &lt;area name&gt; - used to find a walking path
+            to a room with the given name, in the given area if specified
+        set character &lt;name&gt; - sets the current character name (stored as map.character)
+        set recall - sets the current room as the recall room for the current character
+
+    Important Information Regarding Speedwalking
+
+        The existing doSpeedWalk function (which is located at the bottom of the script,
+        just above the event handler function) currently only displays a list of moves
+        to make to get to the target room. This function should be modified to implement
+        speedwalking in your game. For this purpose, a table of roomIDs is generated and
+        and stored in speedWalkPath, and a matching table of directions is stored in
+        speedWalkDir. You can check for doors with a roomID and a direction using the
```

Comment:
May also want to mention `speedWalkWeight` even though it might not be much use to beginning mappers? :grinning:

Reply:
Key question, how does a generic mapper that has no back channel data reliably get the weight for a room while mapping? One could argue movement cost to enter a room, except that in some cases that varies, or is zero (in the game I play and built this script from, there is one cost when walking, a reduced cost when flying, and a zero cost when riding a mount).

In general, I feel that when you have no support from the game, a lot of the features available in the Mudlet mapper can't be used without a much more involved setup that requires fairly extensive tailoring to a game.

Reply:
In some Muds it is possible to classify movement costs by considering the terrain and apply a weighting to favour the easier to move ones (e.g. *inside* rooms  < outside in *city* < out of town roads/paths < open land < forest < mountain < swimable water) - of course this may not be a precise method but it can be used to encourage speedwalking by the most efficient means and it can sometimes show up surprising short-cuts (provided one is prepared for the heath implications of jumping out an upstairs window or crawling through sewers)...

Yeah - weighting is perhaps an *advanced* topic and it can be fiddly to tune it to a player's precise circumstances so if is mentioned here it should only be as a minor aside...!